dnf-repoconfig-daemon/configuration.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "configuration.hpp"
+
+#include "../libdnf/dnf-context.h" // for find_base_arch
+#include "../libdnf/log.hpp"
+#include "../libdnf/utils/os-release.hpp" // for getOsReleaseData
+#include "../libdnf/utils/tinyformat/tinyformat.hpp"
+
+#include <filesystem>
+#include <glob.h>
+#include <iostream>
+#include <rpm/rpmlib.h>
+
+void Configuration::read_configuration()
+{
+    set_substitutions();
+    read_main_config();
+    read_repo_configs();
+}
+
+void Configuration::set_substitutions()
+{
+    const char *arch;
+    rpmGetArchInfo(&arch, NULL);
+    substitutions["arch"] = std::string(arch);
+
+    const auto basearch = find_base_arch(arch);
+    substitutions["basearch"] = basearch ? std::string(basearch) : "";
+
+    // find out release version from rpmdb with fallback to config files
+    const char* releasever = os_release_from_rpmdb("/");
+    if (!releasever) {
+        auto osdata = libdnf::getOsReleaseData();
```

Comment:
os_release_from_rpmdb() looks only for package that provide `system-release(releasever)` in rpmdb and returns its version. In case there is no such package, it returns null. So it unfortunately does not read any config file, just rpmdb.